GoogleSignIn/Tests/Unit/GIDVerifyAccountDetailTest.m

```diff
@@ -0,0 +1,144 @@
+  XCTAssertNil(verifyAccountDetail.configuration.hostedDomain);
+  XCTAssertNil(verifyAccountDetail.configuration.openIDRealm);
+}
+
+- (void)testInit_noConfig {
+  [_fakeMainBundle startFakingWithClientID:kClientId];
+  [_fakeMainBundle fakeWithClientID:nil
+                     serverClientID:nil
+                       hostedDomain:nil
+                        openIDRealm:nil];
+  GIDVerifyAccountDetail *verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+
+  XCTAssertNil(verifyAccountDetail.configuration);
+}
+
+
+- (void)testInit_fullConfig {
+  [_fakeMainBundle startFakingWithClientID:kClientId];
+  [_fakeMainBundle fakeWithClientID:kClientId
+                     serverClientID:kServerClientId
+                       hostedDomain:kFakeHostedDomain
+                        openIDRealm:kOpenIDRealm];
+
+  GIDVerifyAccountDetail *verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+  XCTAssertNotNil(verifyAccountDetail.configuration);
+  XCTAssertEqual(verifyAccountDetail.configuration.clientID, kClientId);
+  XCTAssertEqual(verifyAccountDetail.configuration.serverClientID, kServerClientId);
+  XCTAssertEqual(verifyAccountDetail.configuration.hostedDomain, kFakeHostedDomain);
+  XCTAssertEqual(verifyAccountDetail.configuration.openIDRealm, kOpenIDRealm);
+}
+
+- (void)testInit_invalidConfig {
+  [_fakeMainBundle startFakingWithClientID:kClientId];
+  [_fakeMainBundle fakeWithClientID:@[ @"bad", @"config", @"values" ]
+                     serverClientID:nil
+                       hostedDomain:nil
+                        openIDRealm:nil];
+
+  GIDVerifyAccountDetail *verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+  XCTAssertNil(verifyAccountDetail.configuration);
+}
+
+- (void)testPresentingViewControllerException {
+  _presentingViewController = nil;
+
+  XCTAssertThrows([_verifyAccountDetail verifyAccountDetails:_verifiableAccountDetails
+                                    presentingViewController:_presentingViewController
+                                                  completion:nil]);
+}
+
+- (void)testClientIDMissingException {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnonnull"
+  _verifyAccountDetail.configuration = [[GIDConfiguration alloc] initWithClientID:nil];
+#pragma GCC diagnostic pop
+
+  XCTAssertThrowsSpecificNamed(
+                               [_verifyAccountDetail verifyAccountDetails:_verifiableAccountDetails presentingViewController:_presentingViewController completion:nil],
+                               NSException,
+                               NSInvalidArgumentException,
+                               @"You must specify |clientID| in |GIDConfiguration|");
+}
+
+- (void)testSchemesNotSupportedException {
+  XCTAssertThrowsSpecificNamed(
+                               [_verifyAccountDetail verifyAccountDetails:_verifiableAccountDetails presentingViewController:_presentingViewController completion:nil],
```

Comment:
Formatting: line break and align on the colons.

Reply:
Done.